### PR TITLE
Extend processors_profile_log with more info

### PR DESCRIPTION
**Pull request by amosbird**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Extend processors_profile_log with more information such as input rows. 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Review by kitaisreal — APPROVED** [review]

**Comment by UnamedRus** [comment]
Hm, it's possible to also have memory usage?

**Comment by amosbird** [comment]
> Hm, it's possible to also have memory usage?

I don't think we have per-processor memory tracker. So I guess not for now.

**Comment by kitaisreal** [comment]
@amosbird need to fix tests. Probably issue with different sorting order.

**Review by KochetovNicolai — APPROVED** [review]
LGTM

**Comment by amosbird** [comment]
> Stress test (debug) — Killed by signal (in clickhouse-server.log)
```
/var/log/clickhouse-server/clickhouse-server.err.log:2022.08.15 15:33:43.681159 [ 31258 ] {85e24f33-b06c-4c56-b0b7-74b94a16d997} <Fatal> : Logical error: 'Cannot find column _subquery659 in ActionsDAG result'.
/var/log/clickhouse-server/clickhouse-server.err.log:2022.08.15 15:44:30.105900 [ 32547 ] {} <Fatal> BaseDaemon: ########################################
/var/log/clickhouse-server/clickhouse-server.err.log:2022.08.15 15:44:30.416610 [ 32547 ] {} <Fatal> BaseDaemon: (version 22.8.1.1, build id: 03AEDFFC4BFE22D9) (from thread 31258) (query_id: 85e24f33-b06c-4c56-b0b7-74b94a16d997) (query: SELECT status, count() AS all FROM test ARRAY JOIN nested as nestedJoined WHERE (status IN ( SELECT status FROM test ARRAY JOIN nested as nestedJoined GROUP BY status ORDER BY count() DESC LIMIT 10)) AND (id IN ('1', '2')) GROUP BY CUBE(status) LIMIT 100;) Received signal Aborted (6)
```

> Performance Comparison [1/4] — Failed to parse the report.

```
curl: (6) Could not resolve host: s3.amazonaws.com
```

These errors aren't related.

**Review by kitaisreal — APPROVED** [review]

File: src/Interpreters/ProcessorsProfileLog.cpp [hunk 1]
```diff
@@ -26,12 +26,18 @@ NamesAndTypesList ProcessorProfileLogElement::getNamesAndTypes()
 
         {"id", std::make_shared<DataTypeUInt64>()},
         {"parent_ids", std::make_shared<DataTypeArray>(std::make_shared<DataTypeUInt64>())},
+        {"plan_step", std::make_shared<DataTypeUInt64>()},
+        {"plan_group", std::make_shared<DataTypeUInt64>()},
```
Comment (nudles) [inline src/Interpreters/ProcessorsProfileLog.cpp:30]: Hi @amosbird, may I know the purpose of adding plan_step into the log? Can we use it to get additional information about the processor, e.g., the table/column processed by it? Thank you!

Comment (amosbird) [reply src/Interpreters/ProcessorsProfileLog.cpp:30]: It's used to identify processors as a group. See https://github.com/ClickHouse/ClickHouse/blob/master/src/QueryPipeline/printPipeline.h#L69 . Though it's not perfect.

**Pull request merged**